### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by jeffreyhanson**
@dimitri-justeau, I had a few ideas for improving the README. Could you please take a look and let me know what you think? Briefly, here's a list of the changes.

1. I edited the overview of the package (i.e. first paragraph of the README) to focus more on describing what people can do with the package, rather than describing its implementation. I think something like this will be helpful so that readers get an idea of what they can do with it. Let me know if there's any additional functionality worth highlighting here?
2. I tried shortening the example in the README to provide a brief overview. Since we'll have the vignette to provide a longer tutorial, I suggest making the example more concise so readers can quickly see what the package does.
3. I removed the devtools code for installing the package from GitHub. This is because `devtools::install_github()` is just a wrapper for `remotes::install_github()` - so I figure it's easier just to tell people to use `remotes::install_github()` so they don't have to wait ages for devtools (and all its dependencies) to finish installing.
4. I moved the instructions for compiling the `restopt` core Java library to the end of the installation instructions, because they are optional and most users won't need/use them (in my opinion).
5. I removed the references section - I think it's fine to just have links to the papers in the README. I suggest we have the  have citations/references in the package vignette.
6. I tried editing other text in the README to improve clarity - plese let me know if anything I've edited is now incorrect?
7. (edited) I removed the Table of contents since this will be automatically included in the package website as a sidebar.

### CODE REVIEW

**Comment by jeffreyhanson** [comment]
Ah, I'm sorry - I just remembered that we talked about removing the URLs from the documentation to avoid issues with CRAN's URL checks. I'll remove them now.

**Comment by jeffreyhanson** [comment]
Woops - clicked wrong button.

**Review by dimitri-justeau — COMMENTED** [review]
That's awesome Jeff thank you! I just have two suggestions:

- If we cannot put link to the articles, shouldn't we keep the References section at the end ? I think it will help users that want to see the articles we are referring to.

- I think it would be nice to add just a little sentence describing the test dataset (I will make a suggestion).

File: README.Rmd [hunk 1]
```diff
@@ -148,187 +71,136 @@ brew install openjdk
...
+# load package
+library(restoptr)
 ```
 
-Then, we can add constraints to this base problem. For instance, lets add a
-locked-out constraint, to restrict the number of planning units that can be
-selected for restoration. Such a constraint can be used to account for existing
-land-use practices, feasibility of restoration activities, and stakeholder 
-preferences.
+To identify priorities for restoration, we require information on the spatial location of places that do and do not currently contain suitable habitat. We will now import an example dataset to specify such information (imported as the `habitat_data` object). Specifically, this object is a spatial grid (i.e., raster layer). Each grid cell corresponds to a candidate place for restoration (termed planning unit), and their values indicate the absence or presence of habitat within each planning unit (using values of zero and one, respectively).
```
Comment (dimitri-justeau) [inline README.Rmd]: Maybe before going into the tutorial part, we should provide a sentence to describe the context of the dataset something like:

"An example dataset has been included in the package, it corresponds to the forest cover of a protected area in New Caledonia. This dataset is the one that was used in Justeau-Allaire et al. (2021)."

File: README.Rmd [hunk 2]
```diff
@@ -148,187 +71,136 @@ brew install openjdk
...
+plot(habitat_data, plg = list(x = "topright"))
 ```
 
-And finally a compactness constraint, which limits the spatial extent of the
-selected restoration area:
-
-```{r}
-p <- p %>% add_compactness_constraint(2.4, unit = "km")
-```
+Restoration efforts are often limited in terms of the places where they can be implemented. For instance, restoration efforts may not be feasible in dense cities. To specify which planning units are not feasible for restoration, we will import another example dataset (imported as the `locked_out_data` object). The grid cell values in this raster layer indicate which planning units should be considered available for restoration or not (using values of zero and one, respectively).
```
Comment (dimitri-justeau) [inline README.Rmd]: Just like the previous comment, maybe we could consider adding a context sentence:

"In our example dataset, locked out areas correspond to areas that cannot be accessed using the existing track network"

File: README.Rmd (hunk 1 above)
Comment (jeffreyhanson) [reply README.Rmd]: Yeah, that's an excellent idea!

File: README.Rmd (hunk 2 above)
Comment (jeffreyhanson) [reply README.Rmd]: Awesome - I'll add that in too.

**Comment by jeffreyhanson** [comment]
Ok - personally, I don't think the reference section adds much value - but I've added it back in.

File: README.Rmd (hunk 2 above)
Comment (jeffreyhanson) [reply README.Rmd]: Done!

File: README.Rmd (hunk 1 above)
Comment (jeffreyhanson) [reply README.Rmd]: Done!

**Comment by jeffreyhanson** [comment]
I'll merge this now and then continue on the vignette, but please feel free to edit as needed.

**Comment by dimitri-justeau** [comment]
Thank you ! That's awesome! Do you think small thematic vignettes explaining more in detail specific features (e.g. the data preprocessing process, the optimality gap, etc) would be useful? If so, I can also work on this!

**Comment by dimitri-justeau** [comment]
> Ok - personally, I don't think the reference section adds much value - but I've added it back in.

Okay, please feel free to remove it back if you prefer!

**Comment by jeffreyhanson** [comment]
Thanks! Yeah, I think a paragraph on data pre-processing would be extremely useful (i.e. covering describing why the aggregation stuff is useful and what the parameters do). Once I'm happy to with a draft for the vignette, I'll open a PR and then maybe you could take a look and add in the paragraph on the data pre-processing - how does that sound?

**Comment by dimitri-justeau** [comment]
> Thanks! Yeah, I think a paragraph on data pre-processing would be extremely useful (i.e. covering describing why the aggregation stuff is useful and what the parameters do). Once I'm happy to with a draft for the vignette, I'll open a PR and then maybe you could take a look and add in the paragraph on the data pre-processing - how does that sound?

That sounds great! Do you think we should do one extensive vignette covering most features, several thematic vignettes detailing potential use cases or specific features, or both ?

**Comment by jeffreyhanson** [comment]
Yeah that's a good point! I think several thematic vignettes would be ideal - but this would be more work. If you're happy to lead a vignette dedicated to the aggregation stuff, then that would be awesome? I'm sorry, I don't think I have time to write up a second vignette too.

**Comment by dimitri-justeau** [comment]
Don't worry, the vignette you have started to draft is perfect! I can totally take care of writing a vignette dedicated to aggregation, and maybe we can link to that vignette in yours afterwards? Also I think I will make a vignette to summarize and describe the constraints, and one for the objectives. Then we can put them into a specific menu item in the website, just like I did here: https://dimitri-justeau.github.io/rflsgen/articles/Targets.html . I can also take time to set up a "FAQ" vignette, like this one: https://dimitri-justeau.github.io/rflsgen/articles/FAQ.html

As such specific vignettes are mainly useful for the website, I think they are not prerequisites to CRAN's submission.

**Comment by dimitri-justeau** [comment]
Or maybe, if I do vignettes summarizing constraints and objectives, they should just link to the functions documentation? Like you did here: https://prioritizr.net/reference/index.html (or maybe it was an automatic feature of pkgdown ?)

**Comment by jeffreyhanson** [comment]
Yeah - that all sounds great! I really like the idea of an FAQ vignette. Yeah, you can organize the functions into topics using 
a pkgdown YAML configuration file (e.g., see https://github.com/prioritizr/prioritizr/blob/master/_pkgdown.yml#L75-L79). To create arbitrary help pages that are not about a specific function (e.g., one to outline all the constraints), this can be done using Roxygen2 (e.g., here's prioritizr's constraints help file, https://github.com/prioritizr/prioritizr/blob/master/R/constraints.R).

**Pull request merged**